xcode/Subconscious/Shared/Services/DatabaseService.swift

```diff
@@ -127,39 +208,41 @@ final class DatabaseService {
                 size=excluded.size
             """,
             parameters: [
-                .text(String(describing: entry.slug)),
-                .text(entry.contents.contentType),
-                .date(entry.contents.created),
-                .date(entry.contents.modified),
-                .text(entry.contents.title),
-                .text(entry.contents.fileExtension),
-                .json(entry.contents.headers, or: "[]"),
-                .text(entry.contents.body),
-                .text(entry.contents.plain()),
-                .text(entry.contents.excerpt()),
-                .json(entry.contents.slugs(), or: "[]"),
-                .integer(info.size)
+                .text(address.description),
+                .text(address.slug.description),
+                .text(address.audience.rawValue),
+                .text(memo.contentType),
+                .date(memo.created),
+                .date(memo.modified),
+                .text(memo.title),
+                .text(memo.fileExtension),
+                .json(memo.headers, or: "[]"),
+                .text(memo.body),
+                .text(memo.plain()),
+                .text(memo.excerpt()),
+                .json(memo.slugs(), or: "[]"),
+                .integer(size)
             ]
         )
     }
     
     /// Delete entry from database
-    func removeEntry(slug: Slug) throws {
+    func removeMemo(_ address: MemoAddress) throws {
         guard self.state == .ready else {
             throw DatabaseServiceError.notReady
         }
         try database.execute(
             sql: """
-            DELETE FROM memo WHERE slug = ?
+            DELETE FROM memo WHERE id = ?
             """,
             parameters: [
-                .text(slug.description)
+                .text(address.description)
             ]
         )
     }
     
     /// Count all entries
-    func countEntries() -> Int? {
+    func countMemos() -> Int? {
```

Comment:
Prefer "memo" terminology wherever it makes sense. We used to use "entry" as a generic term.  Now we try to speak concretely about "memos" (an actual data structure) and "entries" (another data structure). Some APIs still use "entry" in the generic legacy sense. We'll try to fix those as we go.